src/toxcore/dht/packet/dht_request.rs

```diff
@@ -264,9 +272,70 @@ impl ToBytes for DhtPkAnnounce {
     }
 }
 
+/** Hardening nodes request of DHT Request packet.
+
+Length    | Content
+--------- | -------------------------
+`1`       | `0x30`
+`1`       | `0x02`
+`rest`    | ignored
+
+*/
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct HardeningRequest;
+
+impl FromBytes for HardeningRequest {
+    named!(from_bytes<HardeningRequest>, do_parse!(
+        tag!("\x30") >>
+        tag!("\x02") >>
+        rest >>
```

Comment:
It's not necessary to parse `rest` since we don't use it and don't require `eof`.